### PR TITLE
initial commit for auth and user service

### PR DESCRIPTION
**Pull request by xx-db**
**PR Checklist**

- [ ] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

### CODE REVIEW

File: server/src/main/java/io/unitycatalog/server/service/PrivilegeAssignment.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package io.unitycatalog.server.service;
+
+import io.unitycatalog.server.model.Privilege;
+import java.util.List;
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+@Builder
+class PrivilegeAssignment {
```
Comment (vikrantpuppala) [inline server/src/main/java/io/unitycatalog/server/service/PrivilegeAssignment.java]: should we define this in the spec and let it be autogenerated?

File: server/src/main/java/io/unitycatalog/server/service/UpdateAuthorizationResponse.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package io.unitycatalog.server.service;
+
+import java.util.List;
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+@Builder
+public class UpdateAuthorizationResponse {
+  List<PrivilegeAssignment> privilege_assignments;
```
Comment (vikrantpuppala) [inline server/src/main/java/io/unitycatalog/server/service/UpdateAuthorizationResponse.java]: camel case

File: server/src/main/java/io/unitycatalog/server/service/UpdateAuthorizationResponse.java
Comment (vikrantpuppala) [inline server/src/main/java/io/unitycatalog/server/service/UpdateAuthorizationResponse.java:1]: move to openapi spec and autogenerate?

File: server/src/main/java/io/unitycatalog/server/service/UpdateAuthorizationRequest.java
Comment (vikrantpuppala) [inline server/src/main/java/io/unitycatalog/server/service/UpdateAuthorizationRequest.java:1]: move to openapi spec and autogenerate?

File: server/src/main/java/io/unitycatalog/server/service/PrivilegeAssignment.java (hunk 1 above)
Comment (creechy) [reply server/src/main/java/io/unitycatalog/server/service/PrivilegeAssignment.java]: Ya. Not a bad idea.

**Pull request merged**